src/Chainweb/Pact/Service/PactInProcApi.hs

```diff
@@ -89,22 +90,29 @@ withPactService'
     -> Maybe FilePath
     -> Maybe NodeId
     -> Bool
-    -> (TQueue RequestMsg -> IO a)
+    -> (PactQueue -> IO a)
     -> IO a
-withPactService' ver cid logger memPoolAccess cdbv bhDb pdb dbDir nodeid
-                 resetDb action =
-    mask $ \rst -> do
-        reqQ <- atomically (newTQueue :: STM (TQueue RequestMsg))
-        a <- withLink $
-             PS.initPactService ver cid logger reqQ memPoolAccess cdbv bhDb
-                                pdb dbDir nodeid resetDb
-        evaluate =<< rst (action reqQ) `finally` closeQueue reqQ `finally` wait a
+withPactService' ver cid logger memPoolAccess cdbv bhDb pdb dbDir nodeid resetDb action = do
+    reqQ <- atomically $ newTBQueue pactQueueSize
+    race (server reqQ) (client reqQ) >>= \case
+        Left () -> error "pact service terminated unexpectedly"
+        Right a -> return a
```

Comment:
Do we ever expect this to exit?

Reply:
Only the client (right side). I think we have test cases that just terminate and expect the server to shut down.